sympy/polys/series/flint_powerseriesring.py

```diff
@@ -0,0 +1,802 @@
+            raise ValueError("Power series precision must be non-negative")
+        self._prec = prec
+
+    def __repr__(self) -> str:
+        return (
+            f"Flint Power Series Ring over {self._domain} with precision {self._prec}"
+        )
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, FlintPowerSeriesRingQQ):
+            return NotImplemented
+        return self._prec == other.prec
+
+    def __hash__(self) -> int:
+        return hash((self._domain, self._prec))
+
+    def __call__(
+        self, coeffs: Sequence[MPQ | int | Sequence], prec: int | None = None
+    ) -> QQSeries:
+        """
+        Create a power series from a list of coefficients.
+
+        If `prec` is not specified, it defaults to the ring's precision.
+        """
+        s: list[MPQ] = []
+        for c in coeffs:
+            if isinstance(c, MPQ):
+                s.append(c)
+            elif isinstance(c, int):
+                s.append(self._domain(c))
+            elif isinstance(c, tuple):
+                s.append(self._domain(*c))
+            else:
+                raise TypeError(f"Unsupported coefficient type: {type(c)}")
+
+        return self.from_list(s, prec)
+
+    @property
+    def domain(self) -> Domain[MPQ]:
+        """Return the ground domain of the power series ring."""
+        return self._domain
+
+    @property
+    def prec(self) -> int:
+        """Return the ring's precision."""
+        return self._prec
+
+    @property
+    def one(self) -> QQSeries:
+        if self._prec == 0:
+            return fmpq_series([1], prec=0)
+        return fmpq_poly([1])
+
+    @property
+    def zero(self) -> QQSeries:
+        if self._prec == 0:
+            return fmpq_series([0], prec=0)
+        return fmpq_poly([0])
+
+    @property
+    def gen(self) -> QQSeries:
+        if self._prec < 2:
+            return fmpq_series([0, 1], prec=self._prec)
+        return fmpq_poly([0, 1])
+
+    def pretty(self, s: QQSeries) -> str:
+        """Return a pretty-printed string representation of a power series."""
+        if isinstance(s, fmpq_poly):
+            return series_pprint(s.coeffs(), None)
+
+        prec = _get_series_precision(s)
+        return series_pprint(s.coeffs(), prec)
+
+    def print(self, s: QQSeries) -> None:
+        """Print a pretty-printed representation of a power series."""
+        print(self.pretty(s))
+
+    def from_list(self, coeffs: list[MPQ], prec: int | None = None) -> QQSeries:
+        """
+        Create a power series from a list of ground coefficients.
+
+        If `prec` is not specified, it defaults to the ring's precision.
+        """
+        if prec is None:
+            if len(coeffs) <= self._prec:
+                return fmpq_poly(coeffs)
+            prec = self._prec
+
+        return fmpq_series(coeffs, prec=prec)
+
+    def to_list(self, s: QQSeries) -> list[MPQ]:
+        """Returns the list of series coefficients."""
+        return s.coeffs()
```

Comment:
It has just occurred to me that as well as having `to_list()` there needs to be a function that gets the precision of the series. I guess it is easy to forget this (python-flint did as well).

I'm not sure what makes sense here like having `R.prec` for the ring precision and `R.get_prec(s)` for the series precision? Maybe even `to_list()` should return the prec as well in a tuple?

Is there any code that uses this function? Does it also need the prec?